app/models/access_request.rb

```diff
@@ -0,0 +1,51 @@
+# == Schema Information
+#
+# Table name: access_request
+#
+#  id               :integer          not null, primary key
+#  email_address    :text
+#  first_name       :text
+#  last_name        :text
+#  organisation     :text
+#  reason           :text
+#  request_date_utc :datetime         not null
+#  requester_id     :integer
+#  status           :integer          not null
+#  requester_email  :text
+#
+
+class AccessRequest < ApplicationRecord
+  belongs_to :user, foreign_key: :requester_id, primary_key: :id
+
+  enum status: {
+    requested: 0,
+    approved: 1,
+    completed: 2,
+    declined: 3,
+  }
+
+  def admin?(current_user)
+    # should probably be using regex here?
+    admin_users = User.where("email ILIKE (?)", "%education.gov.uk")
+    admin_users.include?(current_user)
+  end
+
+  def update_access(access_request, requesting_user)
```

Comment:
Can this be named differently to give a better idea what is happening here?

Reply:
Also, `requested_user` and `requesting_user` are quite hard to differentiate, are there any better names we can use?

Reply:
This method signature doesn't quite feel right to me, actually.

1. `access_request` and `self` are the same thing, so there is no need to pass it in
2. `requester` could and probably should be a `belongs_to` relationship
3. given the above 2 points, this method doesn't need any arguments (and a better name would probably also help)